docs/IDAM.md

```diff
@@ -0,0 +1,86 @@
+## Keycloak Package
+
+For details on the Keycloak Package see [README.md](../src/keycloak/README.md)
+
+## Configure Keycloak client for UDS Package
+
+This section provides instructions on how to automatically create a Keycloak client for your UDS package.
+
+If your application can be configured with an OIDC `clientId` and `clientSecret`, this is the preferred method for creating a Keycloak client. 
+
+* A secret, named `sso-client-<clientId>`, will be automatically generated. This secret holds the client representation. If desired, you can provide a custom `secretName` to be used instead.
+
+* The required fields for creating a Keycloak client are `name`,`clientId`, and `redirectUris`. Make sure to provide these details accurately to ensure the correct setup of your Keycloak client. 
+
+```yaml
+apiVersion: uds.dev/v1alpha1
+kind: Package
+metadata:
+  name: httpbin
+  namespace: httpbin
+spec:
+  sso:
+    - name: Demo SSO httpbin
+      clientId: uds-core-httpbin
+      redirectUris:
+        - "https://httpbin.uds.dev/login"
+```
+
+## AuthService Package
+
+For details on the AuthService Package see [README.md](../src/authservice/README.md)
+
+## Protecting a UDS Package with AuthService
+To enable authentication for applications that do not have native OIDC configuration, UDS Core can utilize AuthService as an authentication layer.
+
+Follow these steps to protect your application with AuthService:
+
+* Set the `isAuthSvcClient` field to `true` in the `sso` configuration of the Package.
+* Ensure that the pods of the application are labeled with: `protect: keycloak`.
```

Comment:
Throwing a comment here since we discussed offline - think there is a play here for automatically handling this? I think if we did per-workload authn/z and netpols for ingress -> authservice it would be doable? We would have to add `selector` here for the workload to wrap, but otherwise I think this is doable?

Reply:
I'm unsure what the best option is.

Yes, you can do per namespace by just dropping a copy of the RequestAuthentication and AuthorizationPolicy resources into every namespace you want protected, yes.

However there are sometimes reasons to want to skip authservice protection on a few Pods for various reasons. (Something completely cluster-internal only perhaps that breaks when authservice is added.) And there's no way to  opt out of the policies if they have no `matchLabels` (and negated `matchLabels` isn't a thing in Istio).

So I think the safest option is to continue to do `protect: keycloak`. But perhaps a Pepr mutating admission webhook could auto-inject `protect: keycloak` into every Pod unless it's opted out? (unsure where opt-out goes, it is security relevant. Could be the UDS CR for most security, but Pod annotations for opt out is probably fine)

Reply:
@bburky to clarify my thought was per-namespace, with selector on a specific workload. So in essence someone would pass a selector into the SSO config for the pod(s) that they want protected. Then we would generate the applicable authn/z pieces for that namespace + selector. This prevents us from having to do any mutation of `protect: keycloak` which I though would be pretty messy (not every chart exposes labels/annotations, so it gets tough to find an easy place to opt out).

Reply:
Cool. Yes per namespace plus a selector should work great. Some reason I didn't consider providing a customizable selector in the UDS CR or something like that. We could probably make it so if you omit the selector it does everything in the namespace, which is the most common case I think.